### ISSUE TITLE
[Bug] wrong scope when getting token silently

### ISSUE THREAD

**Issue by marvinkurock** [AzureAD/microsoft-authentication-library-for-go#322]
**Which version of MSAL Go are you using?**
v0.51

**Where is the issue?**
* Public client
    * [ ] Device code flow
    * [ ] Username/Password (ROPC grant)
    * [x] Authorization code flow 

**Is this a new or an existing app?**

**What version of Go are you using (`go version`)?**

<pre>
$ go version
</pre>
go version go1.17.5 darwin/arm64

**What operating system and processor architecture are you using (`go env`)?**

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
╰─ go env
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/marvinkurock/Library/Caches/go-build"
GOENV="/Users/marvinkurock/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/marvinkurock/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/marvinkurock/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.17.5"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/marvinkurock/development/go/tmw-cli/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/01/3y5lb3fs17d6g4jkgmt9fm8c0000gn/T/go-build1713909736=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>
I am trying to build a CLI authenticating against a web api.
I created a public app with the scope "guid/scope_name".
When I do 
```go
pubApp.AcquireTokenInteractive(context.Background(), scopes)
```
I get a correct token with the specified scope.
I can test die token against jwt.ms and get the correct scope inside the token and the API works correctly.

Also I have implemented a perstistent cache for the token.
When I wait until the access:_token is expired and try to run the cli again, I try to get a new token with:
```go
if len(accounts) > 0 {
    account := accounts[0]
    fmt.Printf("%v\n", account)
    var scopes = []string{viper.GetString("scope")}
    fmt.Printf("%v\n", scopes)
    result, err := pubApp.AcquireTokenSilent(context.Background(), scopes, public.WithSilentAccount(account))
    if err != nil {
      fmt.Println(err)
      panic("Could not aquire token")
    }
```
It also works and I do get a new token without a browser prompt.
BUT the token has a scope of "User.Read", but I still requested the same scope as initially "guid/scope_name".
Is this a bug or am I doing something wrong?

Update:
I set a proxy and caught the request made to AAD. It requested only scopes „profile openid offline..“ but my scope was not given in die querystring. I did send the request manually with my guid/scope_name added and I got a token with my requested scope. 
so why is aquireTokenSilent ignoring my scope?
I printed it prior to the call and the scope variable is correct :/

**Comment by marvinkurock** [comment]
Did a little more digging and found a potential reason.
in the file
> apps/internal/base/base.go line 273
It uses b.AuthParams for the "Refresh" function.
In all other cases it uses the previously created copy named "authParams" in line 231.
```go
authParams := b.AuthParams // This is a copy, as we dont' have a pointer receiver and authParams is not a pointer.
authParams.Scopes = silent.Scopes
authParams.HomeaccountID = silent.Account.HomeAccountID
authParams.AuthorizationType = silent.AuthorizationType
authParams.UserAssertion = silent.UserAssertion
```
In this copy it is setting the scope the user specified.
I stepped through in the debugger and indeed b.AuthParam has an empty scope array, authParams has my scope set.
And later is just uses the default scopes merged with my scopes. Resulting in only the default scopes. This fits my previous observations.

So I guess the fix would be in line *273* b.AuthParams -> authParams

**Comment by chlowell** [comment]
This is fixed in v0.5.3. Thank you for the issue and PR!
